src/services/document-service.spec.js

```diff
@@ -114,4 +114,191 @@ describe('document-service', () => {
+            title: 'Revision 3',
+            slug: 'rev-3',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated C' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section B' } }
+            ]
+          },
+          {
+            title: 'Revision 4',
+            slug: 'rev-4',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated D' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section B' } }
+            ]
+          },
+          {
+            title: 'Revision 5',
+            slug: 'rev-5',
+            sections: [
+              { ...cloneDeep(unrelatedSection), content: { text: 'Unrelated E' } },
+              { ...cloneDeep(sectionToBeDeleted), content: { text: 'Doomed section C' } }
+            ]
+          }
+        ]);
+      });
+
+      describe('and only the second revision is hard-deleted', () => {
+        let documentRevisionsAfterDeletion;
+
+        beforeEach(async () => {
+          await sut.hardDeleteSection({
+            documentKey: documentRevisionsBeforeDeletion[2].key,
+            sectionKey: documentRevisionsBeforeDeletion[2].sections[1].key,
+            sectionRevision: documentRevisionsBeforeDeletion[2].sections[1].revision,
+            reason: 'My reason',
+            deleteAllRevisions: false,
+            user
+          });
+
+          documentRevisionsAfterDeletion = await sut.getAllDocumentRevisionsByKey(documentRevisionsBeforeDeletion[0].key);
+        });
+
+        it('deletes all earlier and later occurrences of that section revision', () => {
+          expect(documentRevisionsAfterDeletion[1].sections[1]).toMatchObject({
+            deletedOn: expect.any(Date),
+            deletedBy: user._id,
+            deletedBecause: 'My reason',
+            content: null
+          });
+          expect(documentRevisionsAfterDeletion[2].sections[1]).toMatchObject({
+            deletedOn: expect.any(Date),
+            deletedBy: user._id,
+            deletedBecause: 'My reason',
+            content: null
+          });
+          expect(documentRevisionsAfterDeletion[3].sections[1]).toMatchObject({
+            deletedOn: expect.any(Date),
+            deletedBy: user._id,
+            deletedBecause: 'My reason',
+            content: null
+          });
+        });
+
+        it('does not delete any other revision of that section', () => {
+          expect(documentRevisionsAfterDeletion[0].sections[1]).toMatchObject({
+            deletedOn: null,
+            deletedBy: null,
+            deletedBecause: null,
+            content: { text: 'Doomed section A' }
+          });
+          expect(documentRevisionsAfterDeletion[4].sections[1]).toMatchObject({
+            deletedOn: null,
+            deletedBy: null,
+            deletedBecause: null,
+            content: { text: 'Doomed section C' }
+          });
+        });
```

Comment:
Maybe we could add a lighter test that checks that none of the other sections are touched at all (not property by property but deep equal)

Reply:
Done